### PR TITLE
refactor(docs): move security page inside smart contract

### LINKED ISSUE #4273: Refactor smart contract security page URLs
**Is your feature request related to a problem? Please describe.**

We have renamed the title of the smart contract security page from `Security` to `Smart contract security` (see https://github.com/ethereum/ethereum-org-website/pull/4212). We should do the same to the file names (currently just 'security') and URLs.

**Describe the solution you'd like**

- Change filenames of smart contract security files
- Update breaking changes (links/anchor references)
- Upload changes to CrowdIn

### PR DESCRIPTION
**Pull request by il3ven**
## Description
- Moved the current [security page](https://ethereum.org/en/developers/docs/security/) into smart contracts.
- Updated links, description and id

The issue also mentioned `Upload changes to CrowdIn`. I am not sure if I need to do this.

## Related Issue

Fixes #4273

### CODE REVIEW

**Comment by minimalsm** [comment]
Hey @il3ven thanks for the very comprehensive PR on this 🤩 Will get round to reviewing this properly today.

> The issue also mentioned Upload changes to CrowdIn. I am not sure if I need to do this.

Nope. Sorry, I should have been more clear. Uploading to CrowdIn is something the core team must do internally.

File: src/intl/en/page-developers-docs.json [hunk 1]
```diff
@@ -73,8 +73,8 @@
   "docs-nav-scaling-sidechains": "Sidechains",
   "docs-nav-scaling-plasma": "Plasma",
   "docs-nav-scaling-validium": "Validium",
-  "docs-nav-security": "Security",
-  "docs-nav-security-description": "Best practices for managing smart contract attacks and vulnerabilitie",
+  "docs-nav-smart-contract-security": "Smart Contract Security",
```
Comment (samajammin) [inline src/intl/en/page-developers-docs.json:76]:
```suggestion
  "docs-nav-smart-contract-security": "Smart contract security",
```
Changing to sentence case for consistency w/ the rest of our links here.

File: src/intl/en/page-developers-docs.json [hunk 2]
```diff
@@ -73,8 +73,8 @@
...
+  "docs-nav-smart-contract-security-description": "Best practices for managing smart contract attacks and vulnerabilitie",
```
Comment (samajammin) [inline src/intl/en/page-developers-docs.json:77]:
```suggestion
  "docs-nav-smart-contract-security-description": "Best practices for managing smart contract attacks and vulnerabilities",
```
Typo fix

File: src/intl/en/page-developers-index.json [hunk 3]
```diff
@@ -64,8 +64,8 @@
   "page-developers-read-docs": "Read the docs",
   "page-developers-scaling-desc": "Solutions for faster transactions",
   "page-developers-scaling-link": "Scaling",
-  "page-developers-security-desc": "Security measures to consider during development",
-  "page-developers-security-link": "Security",
+  "page-developers-smart-contract-security-desc": "Security measures to consider during development of smart contracts",
+  "page-developers-smart-contract-security-link": "Smart Contract Security",
```
Comment (samajammin) [inline src/intl/en/page-developers-index.json:68]:
```suggestion
  "page-developers-smart-contract-security-link": "Smart contract security",
```

File: src/content/contributing/translation-program/content-versions/index.md [hunk 4]
```diff
@@ -89,7 +89,7 @@ Below is a breakdown of the website pages each content bucket contains.
 - [JSON-RPC](/developers/docs/apis/json-rpc/)
 - [Data and analytics](/developers/docs/data-and-analytics/)
 - [Block explorers](/developers/docs/data-and-analytics/block-explorers/)
-- [Security](/developers/docs/security/)
+- [Smart Contract Security](/developers/docs/smart-contracts/security/)
```
Comment (minimalsm) [inline src/content/contributing/translation-program/content-versions/index.md:92]:
```suggestion
- [Smart contract security](/developers/docs/smart-contracts/security/)
```

File: src/content/glossary/index.md [hunk 5]
```diff
@@ -43,7 +43,7 @@ An Application Programming Interface (API) is a set of definitions for how to us
 
 In [Solidity](#solidity), `assert(false)` compiles to `0xfe`, an invalid opcode, which uses up all remaining [gas](#gas) and reverts all changes. When an `assert()` statement fails, something very wrong and unexpected is happening, and you will need to fix your code. You should use `assert()` to avoid conditions that should never, ever occur.
 
-<DocLink to="/developers/docs/security/" title="Security" />
+<DocLink to="/developers/docs/smart-contracts/security/" title="Smart Contract Security" />
```
Comment (minimalsm) [inline src/content/glossary/index.md:46]:
```suggestion
<DocLink to="/developers/docs/smart-contracts/security/" title="Smart contract security" />
```

File: src/content/glossary/index.md [hunk 6]
```diff
@@ -368,7 +368,7 @@ A [transaction](#transaction) sent from a [contract account](#contract-account)
 
 Also known as a "password stretching algorithm," it is used by [keystore](#keystore-file) formats to protect against brute-force, dictionary, and rainbow table attacks on passphrase encryption, by repeatedly hashing the passphrase.
 
-<DocLink to="/developers/docs/security/" title="Security" />
+<DocLink to="/developers/docs/smart-contracts/security/" title="Smart Contract Security" />
```
Comment (minimalsm) [inline src/content/glossary/index.md:371]:
```suggestion
<DocLink to="/developers/docs/smart-contracts/security/" title="Smart contract security" />
```

**Review by minimalsm — CHANGES_REQUESTED** [review]

**Review by minimalsm — APPROVED** [review]

**Comment by minimalsm** [comment]
Updated relevant pages in CrowdIn, thanks for this @il3ven ❤️

This is also your 5th merged PR on the repo so you can claim some new POAPs too! Check [this discord message](https://discord.com/channels/714888181740339261/887696410567385138/894720659077345291) for context.

**Pull request merged**